### PR TITLE
ES-842155 Find and add comments using regex

### PR DESCRIPTION
**Pull request by Venkateshmuruganandam**
Hi All,
UG draft work completed for Find and add comments using regex in DocIO.

Regards
Venkatesh

### CODE REVIEW

File: File-Formats/DocIO/Working-with-Comments.md [hunk 1]
```diff
@@ -139,6 +139,130 @@ document.Close()
...
+            //Iterates through each occurrence and comment it.
+            for (int i = 0; i < textSelection.Count(); i++)
+            {
+                //Get the found text as single text range.
+                WTextRange textRange = textSelection[i].GetAsOneRange();
+                //Get the owner paragraph of the found text.
+                WParagraph paragraph = textRange.OwnerParagraph;
+                //Get the index of the found text.
+                int textIndex = paragraph.ChildEntities.IndexOf(textRange);
+                //Adds comment to a paragraph.
```
Comment (M-A32) [inline File-Formats/DocIO/Working-with-Comments.md:169]: //Add comment to a paragraph.

File: File-Formats/DocIO/Working-with-Comments.md [hunk 2]
```diff
@@ -139,6 +139,130 @@ document.Close()
...
+                comment.Format.UserInitials = "St";
+                //Set the date and time for comment.
+                comment.Format.DateTime = DateTime.Now;
+                //Insert the comment next to the textrange.
+                paragraph.ChildEntities.Insert(textIndex + 1, comment);
+                //Add the paragraph items to the commented items.
+                comment.AddCommentedItem(textRange);
+            }
+        }
+        //Create file stream.
```
Comment (M-A32) [inline File-Formats/DocIO/Working-with-Comments.md:183]: //Create the file stream.

File: File-Formats/DocIO/Working-with-Comments.md [hunk 3]
```diff
@@ -139,6 +139,130 @@ document.Close()
...
+        using (FileStream outputFileStream = new FileStream("Result.docx", FileMode.Create, FileAccess.ReadWrite))
+        {
+            //Save the Word document to file stream.
```
Comment (M-A32) [inline File-Formats/DocIO/Working-with-Comments.md:186]: //Save the Word document to the file stream.

File: File-Formats/DocIO/Working-with-Comments.md [hunk 4]
```diff
@@ -139,6 +139,130 @@ document.Close()
...
+using (WordDocument document = new WordDocument("Input.docx", FormatType.Docx))
+{
+    //Find all occurrence of a particular text end with comma in the document using regex.
+    TextSelection[] textSelection = document.FindAll(new Regex("\\w+,"));
+    if (textSelection != null)
+    {
+        //Iterates through each occurrence and comment it.
+        for (int i = 0; i < textSelection.Count(); i++)
+        {
+            //Get the found text as single text range.
```
Comment (M-A32) [inline File-Formats/DocIO/Working-with-Comments.md:205]: //Get the found text as a single text range.

File: File-Formats/DocIO/Working-with-Comments.md [hunk 5]
```diff
@@ -139,6 +139,130 @@ document.Close()
...
+    using (WordDocument document = new WordDocument(fileStream, FormatType.Docx))
+    {
+        //Find all occurrence of a particular text end with comma in the document using regex.
+        TextSelection[] textSelection = document.FindAll(new Regex("\\w+,"));
+        if (textSelection != null)
+        {
+            //Iterates through each occurrence and comment it.
+            for (int i = 0; i < textSelection.Count(); i++)
+            {
+                //Get the found text as single text range.
```
Comment (M-A32) [inline File-Formats/DocIO/Working-with-Comments.md:163]: //Get the found text as a single text range.

File: File-Formats/DocIO/Working-with-Comments.md [hunk 6]
```diff
@@ -139,6 +139,130 @@ document.Close()
...
+                WParagraph paragraph = textRange.OwnerParagraph;
+                //Get the index of the found text.
+                int textIndex = paragraph.ChildEntities.IndexOf(textRange);
+                //Adds comment to a paragraph.
+                WComment comment = paragraph.AppendComment("comment test_" + i);
+                //Specify the author of the comment.
+                comment.Format.User = "Peter";
+                //Specify the initial of the author.
+                comment.Format.UserInitials = "St";
+                //Set the date and time for comment.
```
Comment (M-A32) [inline File-Formats/DocIO/Working-with-Comments.md:175]: //Set the date and time for the comment.

File: File-Formats/DocIO/Working-with-Comments.md [hunk 7]
```diff
@@ -139,6 +139,130 @@ document.Close()
...
+        //Iterates through each occurrence and comment it.
+        for (int i = 0; i < textSelection.Count(); i++)
+        {
+            //Get the found text as single text range.
+            WTextRange textRange = textSelection[i].GetAsOneRange();
+            //Get the owner paragraph of the found text.
+            WParagraph paragraph = textRange.OwnerParagraph;
+            //Get the index of the found text.
+            int textIndex = paragraph.ChildEntities.IndexOf(textRange);
+            //Adds comment to a paragraph.
```
Comment (M-A32) [inline File-Formats/DocIO/Working-with-Comments.md:211]: //Add comment to a paragraph.

File: File-Formats/DocIO/Working-with-Comments.md [hunk 8]
```diff
@@ -139,6 +139,130 @@ document.Close()
...
+            WComment comment = paragraph.AppendComment("comment test_" + i);
+            //Specify the author of the comment.
+            comment.Format.User = "Peter";
+            //Specify the initial of the author.
+            comment.Format.UserInitials = "St";
+            //Set the date and time for comment.
```
Comment (M-A32) [inline File-Formats/DocIO/Working-with-Comments.md:217]: //Set the date and time for the comment.

File: File-Formats/DocIO/Working-with-Comments.md [hunk 9]
```diff
@@ -139,6 +139,130 @@ document.Close()
...
+If textSelection IsNot Nothing Then
+    ' Iterate through each occurrence and add a comment.
+    For i As Integer = 0 To textSelection.Count() - 1
+        ' Get the found text as a single text range.
+        Dim textRange As WTextRange = textSelection(i).GetAsOneRange()
+        ' Get the owner paragraph of the found text.
+        Dim paragraph As WParagraph = textRange.OwnerParagraph
+        ' Get the index of the found text.
+        Dim textIndex As Integer = paragraph.ChildEntities.IndexOf(textRange)
+        ' Add a comment to a paragraph.
```
Comment (M-A32) [inline File-Formats/DocIO/Working-with-Comments.md:244]: ' Add a comment to the paragraph.

File: File-Formats/DocIO/Working-with-Comments.md [hunk 10]
```diff
@@ -139,6 +139,130 @@ document.Close()
...
+
+{% tabs %}
+{% highlight c# tabtitle="C# [Cross-platform]" %}
+
+using (FileStream fileStream = new FileStream("Input.docx", FileMode.Open, FileAccess.ReadWrite))
+{
+    //Open the existing Word document.
+    using (WordDocument document = new WordDocument(fileStream, FormatType.Docx))
+    {
+        //Find all occurrence of a particular text end with comma in the document using regex.
```
Comment (M-A32) [inline File-Formats/DocIO/Working-with-Comments.md:156]: //Find all occurrence of a particular text ending with comma in the document using regex.

File: File-Formats/DocIO/Working-with-Comments.md [hunk 11]
```diff
@@ -139,6 +139,130 @@ document.Close()
...
+    }               
+}
+
+{% endhighlight %}
+{% highlight c# tabtitle="C# [Windows-specific]" %}
+
+//Open the existing Word document.
+using (WordDocument document = new WordDocument("Input.docx", FormatType.Docx))
+{
+    //Find all occurrence of a particular text end with comma in the document using regex.
```
Comment (M-A32) [inline File-Formats/DocIO/Working-with-Comments.md:198]: //Find all occurrence of a particular text ending with comma in the document using regex.

**Review by M-A32 — CHANGES_REQUESTED** [review]

**Review by Suriya-Balamurugan — APPROVED** [review]
I don't have any concerns in this code changes.

**Review by MohanaselvamJothi — APPROVED** [review]

**Pull request merged**